core/src/main/java/org/mule/runtime/core/api/policy/OperationPolicyProcessor.java

```diff
@@ -65,27 +68,33 @@ public BaseEvent process(BaseEvent operationEvent) throws MuleException {
 
   @Override
   public Publisher<BaseEvent> apply(Publisher<BaseEvent> publisher) {
-    return from(publisher).then(operationEvent -> {
-      PolicyStateId policyStateId = new PolicyStateId(operationEvent.getContext().getCorrelationId(), policy.getPolicyId());
-      Optional<BaseEvent> latestPolicyState = policyStateHandler.getLatestState(policyStateId);
-      BaseEvent variablesProviderEvent =
-          latestPolicyState.orElseGet(() -> BaseEvent.builder(operationEvent.getContext()).message(of(null)).build());
-      policyStateHandler.updateState(policyStateId, variablesProviderEvent);
-      BaseEvent policyEvent = policyEventConverter.createEvent(operationEvent, variablesProviderEvent);
-      Processor operationCall = buildOperationExecutionWithPolicyFunction(nextProcessor, operationEvent);
-      policyStateHandler.updateNextOperation(policyStateId.getExecutionIdentifier(), operationCall);
-      return executePolicyChain(operationEvent, policyStateId, policyEvent);
-    });
+    return from(publisher)
+        .cast(PrivilegedEvent.class)
```

Comment:
Are we completely sure there's no way a BaseEvent can be send as parameter?. Consider all other casts in the code

Reply:
BaseEvent is just an interface, as well as PrivilegedEvent. The implementation is the same in either case.
That being said, there is https://www.mulesoft.org/jira/browse/MULE-13494 pending to review these casts.